subsystem/elevator.py

```diff
@@ -0,0 +1,204 @@
+)
+
+
+class Elevator(Subsystem):  # elevator class
+    motor_extend: SparkMax = SparkMax(
+        config.elevator_motor_extend_id, config=ELEVATOR_CONFIG)  # motor that extends the arm
+    secondary_rotation_motor: SparkMax = SparkMax(
+        config.elevator_secondary_rotation_motor_id, inverted=True, config=SHOLDER_CONFIG)  # motor that rotates the arm
+    main_rotation_motor: SparkMax = SparkMax(
+        config.elevator_main_rotation_motor_id, inverted=False, config=SHOLDER_CONFIG)  # motor that rotates the arm
+    brake: wpilib.Solenoid = wpilib.Solenoid(
+        1, wpilib.PneumaticsModuleType.REVPH, config.elevator_brake_id)  # brake that holds the arm in place
+    initialized: bool = False
+    elevator_bottom_sensor = rev.CANSparkMax(config.elevator_main_rotation_motor_id, rev.CANSparkMax.MotorType.kBrushless).getReverseLimitSwitch(
+        rev.SparkMaxLimitSwitch.Type.kNormallyOpen)
+    elevator_top_sensor = rev.CANSparkMax(config.elevator_main_rotation_motor_id, rev.CANSparkMax.MotorType.kBrushless).getForwardLimitSwitch(
+        rev.SparkMaxLimitSwitch.Type.kNormallyOpen)
+    turn_sensor = rev.CANSparkMax(config.elevator_main_rotation_motor_id,
+                                  rev.MotorType.kBrushless).getAnalog()  # senses the rotation of the elevator
+    brake_enabled: bool = False
+    pose = Pose3d(0, 0, constants.pivot_point_height, 0, 0, 0)
+    elevator_length: float # the length of the elevator in meters
+    rotation_overide: bool = False  # if true, the rotation limits will be ignored
+    extension_overide: bool = False  # if true, the extension limits will be ignored
+    brake_overide: bool = False  # if true, the brake will not be enabled or disabled
+    intake_up: bool = True
+
+    def enable_brake(self):
+        if self.brake_overide:
+            return
+        else:
+            self.brake_enabled = True
+            self.brake.set(True)
+
+    def disable_brake(self):
+        if self.brake_overide:
+            return
+        else:
+            self.brake_enabled = False
+            self.brake.set(False)
+
+    def get_pose(self):
+        # returns the pose of the arm
+        return self.pose
+
+    def set_pose(self, pose: Pose3d):
+        self.pose = pose
+
+    def is_at_position(self, position: Pose3d):
+        if (self.get_pose() == position):
+            return True
+        else:
+            return False
+
+    def init(self):  # initializing motors
+        self.motor_extend.init()
+        self.secondary_rotation_motor.init()
+        self.main_rotation_motor.init()
+        self.secondary_rotation_motor.follow(
+            config.elevator_main_rotation_motor_id, True)
+        self.enable_brake()
+
+    def stop(self):
+        self.motor_extend.set_raw_output(0)
+        self.main_rotation_motor.set_raw_output(0)
+
+    def hard_stop(self):
+        self.stop()
+        self.enable_brake()
+
+    def boundary_box(self, angle: float): #TESTED AND WORKING
```

Comment:
How was it tested? This code passes _some_ tests :-P https://xkcd.com/221/

Reply:
Also, without _automated_ tests you can't protect against regressions. Even if it's just putting down a few of the things you tested by hand in code, it's worth it.